src/rdb/rdb_raft.c

```diff
@@ -2406,6 +2407,7 @@ rdb_raft_start(struct rdb *db)
 		goto err_raft;
 
 	d_rank_list_free(db->d_replicas);
+	db->d_replicas = NULL;
 	rc = rdb_raft_load_replicas(db, db->d_lc_record.dlr_tail - 1);
 	if (rc != 0 && rc != -DER_NONEXIST)
 		goto err_lc;
```

Comment:
If rdb_raft_load_replicas() fails at rdb_lc_lookup() then it will have allocated db->d_replicas and freed db->d_replicas but not assigned it back to NULL. I guess most likely it would be returning DER_NONEXIST in which case the code here will not jump to err_lc. But if it returned something else nonzero, then we goto err_lc and could encounter a double free? If the vulnerability I am imagining is legitimate then I guess rdb_raft_load_replicas() err_replicas: label code could assign d_replicas = NULL after free

Reply:
Indeed. I've added

    db->d_replicas = NULL;

to all places that free db->d_replicas.